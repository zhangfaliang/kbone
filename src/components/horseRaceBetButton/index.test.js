import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import HorseRaceBetButton from './index';

describe('horseRaceBetButton', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls="middle" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixCls null', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixCls default', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls="default" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixCls sp', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls="sp" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('prefixCls withdraw-fore-any', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls="withdraw-fore-any" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('prefixCls withdraw-fore', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls="withdraw-fore" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixCls withdraw-tir-any', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls="withdraw-tir-any" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixCls withdraw-tir', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceBetButton prefixCls="withdraw-tir" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('正常处理', () => {
    expect.assertions(1);
    const wrapper = render(
      <HorseRaceBetButton
        optionName="sp"
        className="test"
        optionId="2.3"
        handicap="4"
        odds="6.7"
        historySp={[]}
        bettingSlipId="65"
        playtypeName="456"
        title="456"
        betLimit="5000"
        playtypeCode="sdf"
        prefixCls="withdraw-tir"
        showOdds={true}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('isErrorSelect', () => {
    expect.assertions(1);
    const wrapper = render(
      <HorseRaceBetButton
        optionName="sp"
        className="test"
        optionId="2.3"
        handicap="4"
        odds="6.7"
        isErrorSelect={true}
        historySp={[]}
        bettingSlipId="65"
        playtypeName="456"
        title="456"
        betLimit="5000"
        playtypeCode="sdf"
        prefixCls="withdraw-tir"
        showOdds={true}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('isHide', () => {
    expect.assertions(1);
    const wrapper = render(
      <HorseRaceBetButton
        optionName="sp"
        className="test"
        optionId="2.3"
        handicap="4"
        odds="6.7"
        isHide={true}
        historySp={[]}
        bettingSlipId="65"
        playtypeName="456"
        title="456"
        betLimit="5000"
        playtypeCode="sdf"
        prefixCls="withdraw-tir"
        showOdds={true}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('disabled', () => {
    expect.assertions(1);
    const wrapper = render(
      <HorseRaceBetButton
        optionName="sp"
        className="test"
        optionId="2.3"
        handicap="4"
        odds="6.7"
        disabled={true}
        historySp={[1, 3, 4]}
        bettingSlipId="65"
        playtypeName="456"
        title="456"
        betLimit="5000"
        playtypeCode="sdf"
        prefixCls="withdraw-tir"
        showOdds={false}
        showBetText={true}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('handleClick', () => {
    expect.assertions(1);
    const button = mount(
      <HorseRaceBetButton
        optionName="sp"
        className="test"
        optionId="2.3"
        handicap="4"
        odds="6.7"
        disabled={false}
        historySp={[]}
        bettingSlipId="65"
        playtypeName="456"
        title="456"
        betLimit="5000"
        playtypeCode="sdf"
        prefixCls="withdraw-tir"
        showOdds={true}
      />
    );
    const selectBetBtn = button.prop('selectBetBtn');
    const click = jest.fn(selectBetBtn);
    button.simulate('click');
    expect(click()).toBe('selectBetBtn');
  });

  it('isActive', () => {
    expect.assertions(1);
    const button = mount(
      <HorseRaceBetButton
        optionName="sp"
        className="test"
        optionId="2.3"
        handicap="4"
        odds="6.7"
        disabled={false}
        historySp={[]}
        bettingSlipId="65"
        playtypeName="456"
        title="456"
        betLimit="5000"
        playtypeCode="sdf"
        prefixCls="withdraw-tir"
        showOdds={true}
      />
    );
    button.setProps({
      isActive: true,
    });
    expect(true).toBe(button.prop('isActive'));
  });
});
