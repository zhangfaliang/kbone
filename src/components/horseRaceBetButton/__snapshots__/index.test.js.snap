// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`horseRaceBetButton base 1`] = `
<div
  class="middle-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton disabled 1`] = `
<div
  class="test withdraw-tir-horse-race-bet-btn disabled"
>
  <span
    class="betText"
  >
    456
  </span>
  <div
    class="historySp"
  >
    <span>
      1
    </span>
    <span>
      3
    </span>
    <span>
      4
    </span>
  </div>
</div>
`;

exports[`horseRaceBetButton isErrorSelect 1`] = `
<div
  class="test withdraw-tir-horse-race-bet-btn isErrorSelect"
>
  <span
    class="odds"
  >
    6.7
  </span>
</div>
`;

exports[`horseRaceBetButton isHide 1`] = `
<div
  class="test withdraw-tir-horse-race-bet-btn disabled"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton prefixCls default 1`] = `
<div
  class="default-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton prefixCls null 1`] = `
<div
  class="null-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton prefixCls sp 1`] = `
<div
  class="sp-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton prefixCls withdraw-fore 1`] = `
<div
  class="withdraw-fore-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton prefixCls withdraw-fore-any 1`] = `
<div
  class="withdraw-fore-any-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton prefixCls withdraw-tir 1`] = `
<div
  class="withdraw-tir-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton prefixCls withdraw-tir-any 1`] = `
<div
  class="withdraw-tir-any-horse-race-bet-btn"
>
  <span
    class="odds"
  />
</div>
`;

exports[`horseRaceBetButton 正常处理 1`] = `
<div
  class="test withdraw-tir-horse-race-bet-btn"
>
  <span
    class="odds"
  >
    6.7
  </span>
</div>
`;
